tests/unit/grains/test_core.py

```diff
@@ -1232,6 +1232,90 @@ def test_lxc_virtual(self):
                             grains.get("virtual"), "container",
                         )
 
+        with patch.object(os.path, "isdir", MagicMock(return_value=False)):
```

Comment:
**suggestion (non-blocking)** `patch` actually creates a MagicMock already. Also it's typically a good idea to `create_autospec`. Also in this case, patching object is probably less-desirable. All-in-all, this could be written like so:

```
with patch("os.path.isdir", autospec=True, return_value=False):
    with patch("os.path.isfile", autospec=True, side_effect=lambda x: x in (...)): 
```
(note, no need to `True if <something> else False` - just `<something>` or `bool(something)` [but, `thing in stuff` produces a bool value anyway)

Reply:
Hi @waynew , 

I just used the `test_lxc_virtual` as basis to create the new tests, that's why the code looks this way. Take a look in lines 1218 to 1223.